### ISSUE TITLE
'unicode' object has no attribute '_meta' when initializing form with QuerySetSequenceModelField from GET/POST

### ISSUE THREAD

**Issue by Lauriy** [yourlabs/django-autocomplete-light#631]
Followed this http://django-autocomplete-light.readthedocs.org/en/master/gfk.html tutorial and generally everything works great, except if the form isn't initialized from the database but from request parameters (invalid form showing errors, MyForm(request.GET)), I get the error mentioned in the title because dal_contenttypes/fields.py prepare_value gets a contenttype-object string ('14-17') as an input and tries to do this:
` return '%s-%s' % (ContentType.objects.get_for_model(value).pk, value.pk)`

To avoid dal from trying to convert '14-17' into '14-17', I have a workaround like this:
```
class CustomQuerySetSequenceModelField(autocomplete.QuerySetSequenceModelField):
    def prepare_value(self, value):
        if isinstance(value, basestring):
            return value

        return super(CustomQuerySetSequenceModelField, self).prepare_value(value)
```

Seems to fix my problem for now. Hoping this is either a one time thing (Google has no answers) and my setup is somehow wrong or that I've been helpful writing this issue.

Here's the full stack trace:

> Environment:
> 
> 
> Request Method: GET
> Request URL: http://192.168.1.81:9009/en/procurements/?category=&transportStart=&transportEnd=&origin=14-17&destination=14-9&buyNowGTE=
> 
> Django Version: 1.8.11
> Python Version: 2.7.10
> Installed Applications:
> ['modeltranslation',
>  'django.contrib.admin',
>  'django.contrib.auth',
>  'django.contrib.contenttypes',
>  'django.contrib.sessions',
>  'django.contrib.messages',
>  'django.contrib.staticfiles',
>  'apps.register',
>  'easy_thumbnails',
>  'apps.common',
>  'apps.location',
>  'apps.insurance',
>  'apps.transport',
>  'apps.payment',
>  'apps.procurement',
>  'django.contrib.sites',
>  'allauth',
>  'allauth.account',
>  'allauth.socialaccount',
>  'allauth.socialaccount.providers.eebankseb',
>  'allauth.socialaccount.providers.eebankswed',
>  'allauth.socialaccount.providers.mobile_id',
>  'allauth.socialaccount.providers.eeidcard',
>  'crispy_forms',
>  'django_forms_bootstrap',
>  'rosetta',
>  'pytz',
>  'rik',
>  'djangoformsetjs',
>  'django_cleanup',
>  'dal',
>  'dal_select2',
>  'dal_queryset_sequence',
>  'static_precompiler',
>  'wkhtmltopdf',
>  'test_without_migrations',
>  'debug_toolbar']
> Installed Middleware:
> ['django.contrib.sessions.middleware.SessionMiddleware',
>  'django.middleware.locale.LocaleMiddleware',
>  'django.middleware.common.CommonMiddleware',
>  'django.middleware.csrf.CsrfViewMiddleware',
>  'apps.register.middlewares.AuthenticationMiddleware',
>  'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
>  'django.contrib.messages.middleware.MessageMiddleware',
>  'django.middleware.clickjacking.XFrameOptionsMiddleware',
>  'django.middleware.security.SecurityMiddleware',
>  'django.middleware.common.BrokenLinkEmailsMiddleware',
>  'debug_toolbar.middleware.DebugToolbarMiddleware']
> 
> 
> Traceback:
> File "/home/lauri/venv/lib/python2.7/site-packages/django/core/handlers/base.py" in get_response
>   164.                 response = response.render()
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/response.py" in render
>   158.             self.content = self.rendered_content
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/response.py" in rendered_content
>   135.         content = template.render(context, self._request)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/backends/django.py" in render
>   74.         return self.template.render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/base.py" in render
>   210.                     return self._render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/test/utils.py" in instrumented_test_render
>   96.     return self.nodelist.render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/base.py" in render
>   905.                 bit = self.render_node(node, context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/base.py" in render_node
>   919.         return node.render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/loader_tags.py" in render
>   135.         return compiled_parent._render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/test/utils.py" in instrumented_test_render
>   96.     return self.nodelist.render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/base.py" in render
>   905.                 bit = self.render_node(node, context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/base.py" in render_node
>   919.         return node.render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/loader_tags.py" in render
>   65.                 result = block.nodelist.render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/base.py" in render
>   905.                 bit = self.render_node(node, context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/base.py" in render_node
>   919.         return node.render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/loader_tags.py" in render
>   65.                 result = block.nodelist.render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/base.py" in render
>   905.                 bit = self.render_node(node, context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/base.py" in render_node
>   919.         return node.render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/defaulttags.py" in render
>   224.                         nodelist.append(node.render(context))
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/base.py" in render
>   959.             output = self.filter_expression.resolve(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/base.py" in resolve
>   675.                 new_obj = func(obj, *arg_vals)
> File "/home/lauri/venv/lib/python2.7/site-packages/crispy_forms/templatetags/crispy_forms_filters.py" in as_crispy_field
>   119.     return template.render(c)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/backends/django.py" in render
>   74.         return self.template.render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/base.py" in render
>   210.                     return self._render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/test/utils.py" in instrumented_test_render
>   96.     return self.nodelist.render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/base.py" in render
>   905.                 bit = self.render_node(node, context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/base.py" in render_node
>   919.         return node.render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/defaulttags.py" in render
>   329.                 return nodelist.render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/base.py" in render
>   905.                 bit = self.render_node(node, context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/base.py" in render_node
>   919.         return node.render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/defaulttags.py" in render
>   329.                 return nodelist.render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/base.py" in render
>   905.                 bit = self.render_node(node, context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/base.py" in render_node
>   919.         return node.render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/defaulttags.py" in render
>   329.                 return nodelist.render(context)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/template/base.py" in render
>   908.             bits.append(force_text(bit))
> File "/home/lauri/venv/lib/python2.7/site-packages/django/utils/encoding.py" in force_text
>   92.                 s = six.text_type(s)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/utils/html.py" in <lambda>
>   390.         klass.__unicode__ = lambda self: mark_safe(klass_unicode(self))
> File "/home/lauri/venv/lib/python2.7/site-packages/django/forms/forms.py" in __str__
>   537.         return self.as_widget()
> File "/home/lauri/venv/lib/python2.7/site-packages/django/forms/forms.py" in as_widget
>   593.         return force_text(widget.render(name, self.value(), attrs=attrs))
> File "/home/lauri/venv/lib/python2.7/site-packages/django/forms/forms.py" in value
>   640.         return self.field.prepare_value(data)
> File "/home/lauri/venv/lib/python2.7/site-packages/dal_contenttypes/fields.py" in prepare_value
>   26.         return '%s-%s' % (ContentType.objects.get_for_model(value).pk,
> File "/home/lauri/venv/lib/python2.7/site-packages/django/contrib/contenttypes/models.py" in get_for_model
>   55.         opts = self._get_opts(model, for_concrete_model)
> File "/home/lauri/venv/lib/python2.7/site-packages/django/contrib/contenttypes/models.py" in _get_opts
>   32.             model = model._meta.concrete_model
> 
> Exception Type: AttributeError at /en/procurements/
> Exception Value: 'unicode' object has no attribute '_meta'

**Comment by jpic** [comment]
Interresting ! This reminds me of what I thought was a hack and didn't want to port to v3 as such:

https://github.com/yourlabs/django-autocomplete-light/blob/v2/autocomplete_light/fields.py#L136-L139

We need to get to the bottom of this, but if we don't then your fix works.

**Comment by jpic** [comment]
Fix in #638 will be in next release B)

Until then, you can use your patch which is compatible ;)

We even got a regression test although it's only a functional one, a unit test would have been cool, if you care please contribute one !!!

Thanks for reporting !

**Comment by jpic** [comment]
Fix released in 3.1.2
